htdocs/product/stock/class/mouvementstock.class.php

```diff
@@ -355,7 +355,7 @@ public function _create($user, $fk_product, $entrepot_id, $qty, $type, $price =
 				$qtyisnotenough = 0;
 				foreach ($product->stock_warehouse[$entrepot_id]->detail_batch as $batchcursor => $prodbatch)
 				{
-					if ($batch != $batchcursor) continue;
+					if ($batch !== $batchcursor) continue; // Do a strict comparison because $batchcursar can be an integer
```

Comment:
What do you thin of this fix that seems better (we want a string comparison whatever is type and not a type comparison):

if ((string) $batch !== (string) $batchcursor)